src/lib.rs

```diff
@@ -1035,16 +1038,136 @@ impl<T: Clone + Integer + Signed> Signed for Ratio<T> {
 }
 
 // String conversions
-impl<T> fmt::Display for Ratio<T>
+impl<T> Display for Ratio<T>
 where
-    T: fmt::Display + Eq + One,
+    T: Display + Clone + Integer,
 {
     /// Renders as `numer/denom`. If denom=1, renders as numer.
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if self.denom.is_one() {
-            write!(f, "{}", self.numer)
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        //write!(f,"{}", self.denom);
+        let non_negative = !(self < &<Ratio<T> as Zero>::zero());
```

Comment:
suggestions for making this check less gross are welcome

Reply:
How about `self >= &Self::zero()` or `*self >= Self::zero()` -- both functionally equivalent, but a little less indirection for the compiler to inline in the latter.

Or to avoid the full complication of `Ord for Ratio<T>`, perhaps use something like `(self.numer >= T::zero()) == (self.denom >= T::zero())`.